src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -270,6 +270,9 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
 
                     $alternatives = array();
                     foreach ($this->services as $key => $associatedService) {
+                        if (isset($this->privates[$key])) {
```

Comment:
What about moving this check into `getServiceIds()` and add an optional `$private = true` argument (which will be deprecated later, in order to the `getServiceIds()` to only return public services ids ?